libsolidity/ast/Types.cpp

```diff
@@ -330,30 +331,48 @@ Type const* Type::fullEncodingType(bool _inLibraryCall, bool _encoderV2, bool) c
 MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _scope)
 {
 	vector<UsingForDirective const*> usingForDirectives;
-	if (auto const* sourceUnit = dynamic_cast<SourceUnit const*>(&_scope))
-		usingForDirectives += ASTNode::filteredNodes<UsingForDirective>(sourceUnit->nodes());
-	else if (auto const* contract = dynamic_cast<ContractDefinition const*>(&_scope))
-		usingForDirectives +=
-			contract->usingForDirectives() +
-			ASTNode::filteredNodes<UsingForDirective>(contract->sourceUnit().nodes());
+	SourceUnit const* sourceUnit = dynamic_cast<SourceUnit const*>(&_scope);
+	if (auto const* contract = dynamic_cast<ContractDefinition const*>(&_scope))
+	{
+		sourceUnit = &contract->sourceUnit();
+		usingForDirectives += contract->usingForDirectives();
+	}
 	else
-		solAssert(false, "");
+		solAssert(sourceUnit, "");
+	usingForDirectives += ASTNode::filteredNodes<UsingForDirective>(sourceUnit->nodes());
 
 	// Normalise data location of type.
 	DataLocation typeLocation = DataLocation::Storage;
 	if (auto refType = dynamic_cast<ReferenceType const*>(&_type))
 		typeLocation = refType->location();
 
-	set<Declaration const*> seenFunctions;
+	set<pair<string, Declaration const*>> seenFunctions;
 	MemberList::MemberMap members;
 
+	auto addFunction = [&](FunctionDefinition const& _function, optional<string> _name = {})
```

Comment:
```suggestion
	auto addFunction = [&, seenFunctions = set<pair<string, Declaration const*>>{}](FunctionDefinition const& _function, optional<string> _name = {})
```
I'd actually consider this - I was wondering where it's actually used for a second, since it's defined outside.
But also fine keeping it as is.

Reply:
I tried it, but it didn't work. I moved the variable closer to the function, though.

Reply:
Ah, the lambda would probably have to be declared ``mutable`` to get a non-const capture. But yeah, doesn't matter.